packages/cli/generation/ir-generator/src/converters/services/convertTransport.ts

```diff
@@ -34,3 +34,59 @@ export async function convertTransport({
         service: protobufService
     });
 }
+
+export async function getTransportForEndpoint({
```

Comment:
I left a few comments based on the current structure, but I wonder if we could simplify this to:
1. If the service's transport is not gRPC, return undefined.
2. If the service's source and endpoint's source don't match, set the endpoint's transport (i.e. it's an override).

That way, it's technically possible for a single Fern service to use multiple gRPC services, or a different combination of transports (e.g. gRPC and HTTP).

Thoughts?